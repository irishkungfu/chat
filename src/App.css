body {
  padding: 20px;
}
html {
  box-sizing: border-box;
}
*, *:before, *:after {
  box-sizing: inherit;
}
::-webkit-scrollbar {
  width: 0;
}