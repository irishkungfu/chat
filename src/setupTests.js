import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

export function configureAdapter() {
    configure({ adapter: new Adapter() });
}