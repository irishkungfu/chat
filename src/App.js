import React, { Component } from 'react';
import './App.css';
import MessageListModel from "./models/MessageListModel"

import ChatWrapper from './components/ChatWrapper/ChatWrapper';
import { Provider } from "mobx-react"
import styled, { ThemeProvider } from "styled-components"

// TODO Move to Own File

// Define our button, but with the use of props.theme this time
const Button = styled.button`
  font-size: 1em;
  margin: 1em;
  padding: 0.25em 1em;
  border-radius: 3px;

  /* Color the border and text with theme.main */
  color: ${props => props.theme.main};
  border: 2px solid ${props => props.theme.main};
`;

// We are passing a default theme for Buttons that arent wrapped in the ThemeProvider
Button.defaultProps = {
  theme: {
    main: "palevioletred"
  }
}

// Define what props.theme will look like
const theme = {
  main: "mediumseagreen"
};

const store = new MessageListModel()


class App extends Component {
  render() {
    return (
      <div className="App">
        <ThemeProvider theme={theme}>
          <Provider store={store}>
            <ChatWrapper store={store} />
          </Provider>
        </ThemeProvider>
      </div>
    );
  }
}

export default App;
