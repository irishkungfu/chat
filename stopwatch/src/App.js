import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import { Provider } from "mobx-react"
import TimerModel from "./models/TimerModel"

const store = new TimerModel()

class App extends Component {
  render() {
    return (
      <Provider store={store}>
        <div className="App">

        </div>
      </Provider>

    );
  }
}

export default App;
