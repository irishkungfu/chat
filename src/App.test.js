import React from 'react';
import App from './App';
import { shallow } from 'enzyme';
import { configureAdapter } from "./setupTests"

configureAdapter()

it('renders without crashing', () => {
  // this is a workaround, since jest can't test scrollIntoView
  Element.prototype.scrollIntoView = jest.fn()

  shallow(<App />);
});

describe('App', () => {
  let wrapper;
  beforeEach(() => wrapper = shallow(<App />));

  it('should render correctly', () => expect(wrapper).toMatchSnapshot());

  it('should render a <div />', () => {
    expect(wrapper.find('div').length).toEqual(1);
  });

  it('should render a Provider', () => {
    expect(wrapper.find("Provider").length).toEqual(1)
  })
  it('should render a ChatWrapper', () => {
    expect(wrapper.find("ChatWrapper").length).toEqual(1)
  })
  it('should render a ThemeProvider', () => {
    expect(wrapper.find("ThemeProvider").length).toEqual(1)
  })
});