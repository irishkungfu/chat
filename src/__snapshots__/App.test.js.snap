// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ThemeProvider
        theme={
          Object {
            "main": "mediumseagreen",
          }
        }
      >
        <Provider
          store={
            MessageListModel {
              "awaitingMessage": "done",
              "chatHeaderHeight": 50,
              "chatInputHeight": 50,
              "messages": Array [],
              "newMessage": "",
              Symbol(mobx did run lazy initializers): true,
              Symbol(mobx administration): ObservableObjectAdministration$$1 {
                "defaultEnhancer": [Function],
                "keysAtom": Atom$$1 {
                  "diffValue": 0,
                  "isBeingObserved": false,
                  "isPendingUnobservation": false,
                  "lastAccessedBy": 0,
                  "lowestObserverState": 2,
                  "name": "MessageListModel@1.keys",
                  "observers": Set {},
                },
                "name": "MessageListModel@1",
                "target": [Circular],
                "values": Map {
                  "messages" => Array [],
                  "newMessage" => "",
                  "awaitingMessage" => "done",
                  "chatInputHeight" => 50,
                  "chatHeaderHeight" => 50,
                },
              },
            }
          }
        >
          <ChatWrapper
            store={
              MessageListModel {
                "awaitingMessage": "done",
                "chatHeaderHeight": 50,
                "chatInputHeight": 50,
                "messages": Array [],
                "newMessage": "",
                Symbol(mobx did run lazy initializers): true,
                Symbol(mobx administration): ObservableObjectAdministration$$1 {
                  "defaultEnhancer": [Function],
                  "keysAtom": Atom$$1 {
                    "diffValue": 0,
                    "isBeingObserved": false,
                    "isPendingUnobservation": false,
                    "lastAccessedBy": 0,
                    "lowestObserverState": 2,
                    "name": "MessageListModel@1.keys",
                    "observers": Set {},
                  },
                  "name": "MessageListModel@1",
                  "target": [Circular],
                  "values": Map {
                    "messages" => Array [],
                    "newMessage" => "",
                    "awaitingMessage" => "done",
                    "chatInputHeight" => 50,
                    "chatHeaderHeight" => 50,
                  },
                },
              }
            }
          />
        </Provider>
      </ThemeProvider>,
      "className": "App",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Provider
          store={
            MessageListModel {
              "awaitingMessage": "done",
              "chatHeaderHeight": 50,
              "chatInputHeight": 50,
              "messages": Array [],
              "newMessage": "",
              Symbol(mobx did run lazy initializers): true,
              Symbol(mobx administration): ObservableObjectAdministration$$1 {
                "defaultEnhancer": [Function],
                "keysAtom": Atom$$1 {
                  "diffValue": 0,
                  "isBeingObserved": false,
                  "isPendingUnobservation": false,
                  "lastAccessedBy": 0,
                  "lowestObserverState": 2,
                  "name": "MessageListModel@1.keys",
                  "observers": Set {},
                },
                "name": "MessageListModel@1",
                "target": [Circular],
                "values": Map {
                  "messages" => Array [],
                  "newMessage" => "",
                  "awaitingMessage" => "done",
                  "chatInputHeight" => 50,
                  "chatHeaderHeight" => 50,
                },
              },
            }
          }
        >
          <ChatWrapper
            store={
              MessageListModel {
                "awaitingMessage": "done",
                "chatHeaderHeight": 50,
                "chatInputHeight": 50,
                "messages": Array [],
                "newMessage": "",
                Symbol(mobx did run lazy initializers): true,
                Symbol(mobx administration): ObservableObjectAdministration$$1 {
                  "defaultEnhancer": [Function],
                  "keysAtom": Atom$$1 {
                    "diffValue": 0,
                    "isBeingObserved": false,
                    "isPendingUnobservation": false,
                    "lastAccessedBy": 0,
                    "lowestObserverState": 2,
                    "name": "MessageListModel@1.keys",
                    "observers": Set {},
                  },
                  "name": "MessageListModel@1",
                  "target": [Circular],
                  "values": Map {
                    "messages" => Array [],
                    "newMessage" => "",
                    "awaitingMessage" => "done",
                    "chatInputHeight" => 50,
                    "chatHeaderHeight" => 50,
                  },
                },
              }
            }
          />
        </Provider>,
        "theme": Object {
          "main": "mediumseagreen",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <ChatWrapper
            store={
              MessageListModel {
                "awaitingMessage": "done",
                "chatHeaderHeight": 50,
                "chatInputHeight": 50,
                "messages": Array [],
                "newMessage": "",
                Symbol(mobx did run lazy initializers): true,
                Symbol(mobx administration): ObservableObjectAdministration$$1 {
                  "defaultEnhancer": [Function],
                  "keysAtom": Atom$$1 {
                    "diffValue": 0,
                    "isBeingObserved": false,
                    "isPendingUnobservation": false,
                    "lastAccessedBy": 0,
                    "lowestObserverState": 2,
                    "name": "MessageListModel@1.keys",
                    "observers": Set {},
                  },
                  "name": "MessageListModel@1",
                  "target": [Circular],
                  "values": Map {
                    "messages" => Array [],
                    "newMessage" => "",
                    "awaitingMessage" => "done",
                    "chatInputHeight" => 50,
                    "chatHeaderHeight" => 50,
                  },
                },
              }
            }
          />,
          "store": MessageListModel {
            "awaitingMessage": "done",
            "chatHeaderHeight": 50,
            "chatInputHeight": 50,
            "messages": Array [],
            "newMessage": "",
            Symbol(mobx did run lazy initializers): true,
            Symbol(mobx administration): ObservableObjectAdministration$$1 {
              "defaultEnhancer": [Function],
              "keysAtom": Atom$$1 {
                "diffValue": 0,
                "isBeingObserved": false,
                "isPendingUnobservation": false,
                "lastAccessedBy": 0,
                "lowestObserverState": 2,
                "name": "MessageListModel@1.keys",
                "observers": Set {},
              },
              "name": "MessageListModel@1",
              "target": [Circular],
              "values": Map {
                "messages" => Array [],
                "newMessage" => "",
                "awaitingMessage" => "done",
                "chatInputHeight" => 50,
                "chatHeaderHeight" => 50,
              },
            },
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "store": MessageListModel {
              "awaitingMessage": "done",
              "chatHeaderHeight": 50,
              "chatInputHeight": 50,
              "messages": Array [],
              "newMessage": "",
              Symbol(mobx did run lazy initializers): true,
              Symbol(mobx administration): ObservableObjectAdministration$$1 {
                "defaultEnhancer": [Function],
                "keysAtom": Atom$$1 {
                  "diffValue": 0,
                  "isBeingObserved": false,
                  "isPendingUnobservation": false,
                  "lastAccessedBy": 0,
                  "lowestObserverState": 2,
                  "name": "MessageListModel@1.keys",
                  "observers": Set {},
                },
                "name": "MessageListModel@1",
                "target": [Circular],
                "values": Map {
                  "messages" => Array [],
                  "newMessage" => "",
                  "awaitingMessage" => "done",
                  "chatInputHeight" => 50,
                  "chatHeaderHeight" => 50,
                },
              },
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ThemeProvider
          theme={
            Object {
              "main": "mediumseagreen",
            }
          }
        >
          <Provider
            store={
              MessageListModel {
                "awaitingMessage": "done",
                "chatHeaderHeight": 50,
                "chatInputHeight": 50,
                "messages": Array [],
                "newMessage": "",
                Symbol(mobx did run lazy initializers): true,
                Symbol(mobx administration): ObservableObjectAdministration$$1 {
                  "defaultEnhancer": [Function],
                  "keysAtom": Atom$$1 {
                    "diffValue": 0,
                    "isBeingObserved": false,
                    "isPendingUnobservation": false,
                    "lastAccessedBy": 0,
                    "lowestObserverState": 2,
                    "name": "MessageListModel@1.keys",
                    "observers": Set {},
                  },
                  "name": "MessageListModel@1",
                  "target": [Circular],
                  "values": Map {
                    "messages" => Array [],
                    "newMessage" => "",
                    "awaitingMessage" => "done",
                    "chatInputHeight" => 50,
                    "chatHeaderHeight" => 50,
                  },
                },
              }
            }
          >
            <ChatWrapper
              store={
                MessageListModel {
                  "awaitingMessage": "done",
                  "chatHeaderHeight": 50,
                  "chatInputHeight": 50,
                  "messages": Array [],
                  "newMessage": "",
                  Symbol(mobx did run lazy initializers): true,
                  Symbol(mobx administration): ObservableObjectAdministration$$1 {
                    "defaultEnhancer": [Function],
                    "keysAtom": Atom$$1 {
                      "diffValue": 0,
                      "isBeingObserved": false,
                      "isPendingUnobservation": false,
                      "lastAccessedBy": 0,
                      "lowestObserverState": 2,
                      "name": "MessageListModel@1.keys",
                      "observers": Set {},
                    },
                    "name": "MessageListModel@1",
                    "target": [Circular],
                    "values": Map {
                      "messages" => Array [],
                      "newMessage" => "",
                      "awaitingMessage" => "done",
                      "chatInputHeight" => 50,
                      "chatHeaderHeight" => 50,
                    },
                  },
                }
              }
            />
          </Provider>
        </ThemeProvider>,
        "className": "App",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Provider
            store={
              MessageListModel {
                "awaitingMessage": "done",
                "chatHeaderHeight": 50,
                "chatInputHeight": 50,
                "messages": Array [],
                "newMessage": "",
                Symbol(mobx did run lazy initializers): true,
                Symbol(mobx administration): ObservableObjectAdministration$$1 {
                  "defaultEnhancer": [Function],
                  "keysAtom": Atom$$1 {
                    "diffValue": 0,
                    "isBeingObserved": false,
                    "isPendingUnobservation": false,
                    "lastAccessedBy": 0,
                    "lowestObserverState": 2,
                    "name": "MessageListModel@1.keys",
                    "observers": Set {},
                  },
                  "name": "MessageListModel@1",
                  "target": [Circular],
                  "values": Map {
                    "messages" => Array [],
                    "newMessage" => "",
                    "awaitingMessage" => "done",
                    "chatInputHeight" => 50,
                    "chatHeaderHeight" => 50,
                  },
                },
              }
            }
          >
            <ChatWrapper
              store={
                MessageListModel {
                  "awaitingMessage": "done",
                  "chatHeaderHeight": 50,
                  "chatInputHeight": 50,
                  "messages": Array [],
                  "newMessage": "",
                  Symbol(mobx did run lazy initializers): true,
                  Symbol(mobx administration): ObservableObjectAdministration$$1 {
                    "defaultEnhancer": [Function],
                    "keysAtom": Atom$$1 {
                      "diffValue": 0,
                      "isBeingObserved": false,
                      "isPendingUnobservation": false,
                      "lastAccessedBy": 0,
                      "lowestObserverState": 2,
                      "name": "MessageListModel@1.keys",
                      "observers": Set {},
                    },
                    "name": "MessageListModel@1",
                    "target": [Circular],
                    "values": Map {
                      "messages" => Array [],
                      "newMessage" => "",
                      "awaitingMessage" => "done",
                      "chatInputHeight" => 50,
                      "chatHeaderHeight" => 50,
                    },
                  },
                }
              }
            />
          </Provider>,
          "theme": Object {
            "main": "mediumseagreen",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <ChatWrapper
              store={
                MessageListModel {
                  "awaitingMessage": "done",
                  "chatHeaderHeight": 50,
                  "chatInputHeight": 50,
                  "messages": Array [],
                  "newMessage": "",
                  Symbol(mobx did run lazy initializers): true,
                  Symbol(mobx administration): ObservableObjectAdministration$$1 {
                    "defaultEnhancer": [Function],
                    "keysAtom": Atom$$1 {
                      "diffValue": 0,
                      "isBeingObserved": false,
                      "isPendingUnobservation": false,
                      "lastAccessedBy": 0,
                      "lowestObserverState": 2,
                      "name": "MessageListModel@1.keys",
                      "observers": Set {},
                    },
                    "name": "MessageListModel@1",
                    "target": [Circular],
                    "values": Map {
                      "messages" => Array [],
                      "newMessage" => "",
                      "awaitingMessage" => "done",
                      "chatInputHeight" => 50,
                      "chatHeaderHeight" => 50,
                    },
                  },
                }
              }
            />,
            "store": MessageListModel {
              "awaitingMessage": "done",
              "chatHeaderHeight": 50,
              "chatInputHeight": 50,
              "messages": Array [],
              "newMessage": "",
              Symbol(mobx did run lazy initializers): true,
              Symbol(mobx administration): ObservableObjectAdministration$$1 {
                "defaultEnhancer": [Function],
                "keysAtom": Atom$$1 {
                  "diffValue": 0,
                  "isBeingObserved": false,
                  "isPendingUnobservation": false,
                  "lastAccessedBy": 0,
                  "lowestObserverState": 2,
                  "name": "MessageListModel@1.keys",
                  "observers": Set {},
                },
                "name": "MessageListModel@1",
                "target": [Circular],
                "values": Map {
                  "messages" => Array [],
                  "newMessage" => "",
                  "awaitingMessage" => "done",
                  "chatInputHeight" => 50,
                  "chatHeaderHeight" => 50,
                },
              },
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "store": MessageListModel {
                "awaitingMessage": "done",
                "chatHeaderHeight": 50,
                "chatInputHeight": 50,
                "messages": Array [],
                "newMessage": "",
                Symbol(mobx did run lazy initializers): true,
                Symbol(mobx administration): ObservableObjectAdministration$$1 {
                  "defaultEnhancer": [Function],
                  "keysAtom": Atom$$1 {
                    "diffValue": 0,
                    "isBeingObserved": false,
                    "isPendingUnobservation": false,
                    "lastAccessedBy": 0,
                    "lowestObserverState": 2,
                    "name": "MessageListModel@1.keys",
                    "observers": Set {},
                  },
                  "name": "MessageListModel@1",
                  "target": [Circular],
                  "values": Map {
                    "messages" => Array [],
                    "newMessage" => "",
                    "awaitingMessage" => "done",
                    "chatInputHeight" => 50,
                    "chatHeaderHeight" => 50,
                  },
                },
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
